src/toil/batchSystems/abstractBatchSystem.py

```diff
@@ -236,6 +239,20 @@ def setOptions(cls, setOption):
            used to update run configuration
         """
         pass
+        
+    def getWorkerContexts(self):
+        """
+        Get a list of picklable context manager objects to wrap worker work in,
+        in order.
+        
+        Can be used to ask the Toil worker to do things in-process (such as
+        configuring environment variables, hot-deploying user scripts, or
+        cleaning up a node) that would otherwise require a wrapping "executor"
+        process.
+        
+        :rtype: list
+        """
+        return []
```

Comment:
Should this raise a NotImplementedError?  Otherwise passing contexts to classes inheriting this might unexpectedly seem to accept a list of contexts that are not used.

Reply:
No, batch systems that don't use the feature ought to return an empty list.

Batch systems that do use the feature will return the contexts that they need the worker to run jobs in. The leader queries the batch system and passes the results along to the worker. If a batch system doesn't override the method, the leader needs to get a result that says that no contexts are needed (i.e. an empty list).

You don't pass the contexts *into* the batch system for it to hold and give back. The batch system is responsible for deciding what contexts are needed.